devolutions-session/src/main.rs

```diff
@@ -21,13 +21,17 @@ fn main() -> anyhow::Result<()> {
 
     let config = ConfHandle::init().context("Failed to initialize configuration")?;
 
-    let _logger_guard = init_log(config.clone());
-
     info!("Starting Devolutions Session");
 
+    #[cfg(not(windows))]
+    let _logger_guard = init_log(config);
+
     // TMP: Copy-paste from MSRDPEX project for testing purposes.
     #[cfg(windows)]
-    loop_dvc(config);
+    {
+        let _logger_guard = init_log(config.clone());
+        loop_dvc(config);
+    }
```

Comment:
This is problematic when building on Windows. The guard will get dropped at the end of the very short scope which is only present so that we can conditionally compile the code. I recommend to revert the code to its previous form, and use the `#[expect()]` attribute to silent the warning.

I would go with something like this:
```rust
    #[cfg_attr(not(windows), expect(clippy::redundant_clone))]
    let _logger_guard = init_log(config.clone());
```

The lint is expected to be fired on platforms other than Windows, and if it ever stops being fired, we’ll get warned about that so we can remove the now-useless attribute.

Reply:
Alternatively, `init_log` can take a reference.

```rs
pub fn init_log(config: &ConfHandle) -> LoggerGuard {
    ..
}

let _logger_guard = init_log(&config);  // same as before, without a `cfg` attribute
```

This would pass the [`needless_pass_by_value`](https://rust-lang.github.io/rust-clippy/master/index.html#/needless_pass) lint.

Reply:
Sounds good to me. Actually, this function could be modified to take a `&Conf`. There is no need for the whole `ConfHandle` which exists to allow hot reloading of the configuration.

```rust
pub fn init_log(config: &Conf) -> LoggerGuard {
    devolutions_log::init::<SessionLog>(
        &conf.log_file,
        conf.verbosity_profile.to_log_filter(),
        conf.debug.log_directives.as_deref(),
    )
    .expect("BUG: Failed to initialize log")
}
```

Reply:
Addressed in 548cfc3.